Lib/importlib/readers.py

```diff
@@ -0,0 +1,30 @@
+import pathlib
+import zipfile
+from . import abc
+
+
+class FileReader(abc.TraversableResources):
+    def __init__(self, loader):
+        self.path = pathlib.Path(loader.path).parent
+
+    def files(self):
+        return self.path
+
+
+class ZipReader(FileReader):
```

Comment:
I'm probably going to decouple FileReader from ZipReader. The only shared functionality is the one-line `files` method.